tensorflow_datasets/image/lfw.py

```diff
@@ -0,0 +1,122 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Labeled faces in wild."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import tensorflow_datasets.public_api as tfds
+
+
+_URL = "http://vis-www.cs.umass.edu/lfw/lfw.tgz"
+
+LFW_IMAGE_SHAPE = (250, 250, 3)
+
+LFW_CITATION = """\
+@TechReport{LFWTech,
+    author = {Gary B. Huang and Manu Ramesh and Tamara Berg and Erik Learned-Miller},
+    title = {Labeled Faces in the Wild: A Database for Studying Face Recognition in Unconstrained Environments},
+    institution = {University of Massachusetts, Amherst},
+    year = 2007,
+    number = {07-49},
+    month = {October}
+}
+"""
+
+
+class LFW(tfds.core.GeneratorBasedBuilder):
+  """LFW Class"""
+  URL = "http://vis-www.cs.umass.edu/lfw/#resources"
+  VERSION = tfds.core.Version("2.0.0")
+  def _info(self):
+    return tfds.core.DatasetInfo(
+        builder=self,
+        description=("Labeled Faces in the Wild:\
+        A Database for Studying Face Recognition in Unconstrained Environments"),
+        features=tfds.features.FeaturesDict({
+            "anchor": tfds.features.Image(shape=LFW_IMAGE_SHAPE),
+            "positive": tfds.features.Image(shape=LFW_IMAGE_SHAPE),
+            "negative": tfds.features.Image(shape=LFW_IMAGE_SHAPE),
+            }),
+        urls=[self.URL],
+        citation=LFW_CITATION,
+        )
+
+  def _split_generators(self, dl_manager):
+    path = dl_manager.download_and_extract(_URL)
+    # There is no train/test split predefined
+    return [
+        tfds.core.SplitGenerator(
+            name=tfds.Split.TRAIN,
+            num_shards=20,
+            gen_kwargs={
+                "data_path": path,}
+            ),
+        ]
+
+  def _generate_examples(self, data_path):
+    print("Generating triplets, this will take a while")
+    # a list of dictionary will be recieved
+    # each element(dict) will have 3 keys
+    # each of which will store the path to the image
+    triplet_list = self.triplet_maker(data_path)
+
+    for triplet in triplet_list:
+      yield {
+          "anchor": triplet["anchor"],
+          "positive": triplet["positive"],
+          "negative": triplet["negative"],
+      }
+
+  #This is a helper function for making all possible triplets for siamese network(eg. FaceNet)
+  def triplet_maker(self, _path):
+    triplet_list = []
+    lfw = os.listdir(_path)
```

Comment:
You shuld use `tf.io.gfile` for all filesystem access. Check this [link](https://stackoverflow.com/a/42936969/8401425) about why using that

Reply:
I understand I have removed them with tf.gfile.ListDirectory

Reply:
Also ran and checked, every thing seems to work fine on my system.